### PR TITLE
Support for dropping malformed rows for TimestampType and DateType types

### PR DESCRIPTION
**Pull request by HyukjinKwon**
In this PR, I added some logics to catch `IllegalArgumentException`. When trying to parse malformed data to `TimestampType` or `DateType`, it emits the exception with the message below. 

```
java.lang.IllegalArgumentException
	at java.sql.Date.valueOf(Date.java:143)
	at com.databricks.spark.csv.util.TypeCast$.castTo(TypeCast.scala:64)
	at com.databricks.spark.csv.CsvRelation$$anonfun$buildScan$6.apply(CsvRelation.scala:188)
	at com.databricks.spark.csv.CsvRelation$$anonfun$buildScan$6.apply(CsvRelation.scala:169)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:396)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:369)
	...
```

So, because of this, dropping malformed rows does not work properly.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
I am worried of few things.

Firstly, I did not want to add another CSV file and test separately to check this so I just correct original tests and the data. If this is inappropriate, I will get that back and add another test.

Secondly, for me `IllegalArgumentException` and the message `Illegal argument exception` does not sound like it failed during parsing. Would you tell me anything you would prefer if you feel in the same way?

File: src/main/scala/com/databricks/spark/csv/CsvRelation.scala [hunk 1]
```diff
@@ -129,7 +129,13 @@ case class CsvRelation protected[spark] (
               s"Dropping malformed line: ${tokens.mkString(",")}")
             None
           case pe: java.text.ParseException if dropMalformed =>
```
Comment (falaki) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: You can catch both exceptions together:
```Scala
case _ : java.lang.NumberFormatException | _ : IllegalArgumentException =>
```

Comment (falaki) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: Also please add this new exception handling case to `buildScan(requiredColumns: Array[String])` as well.

Comment (HyukjinKwon) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: I definitely will.

File: src/main/scala/com/databricks/spark/csv/CsvRelation.scala [hunk 2]
```diff
@@ -124,13 +124,14 @@ case class CsvRelation protected[spark] (
           case aiob: ArrayIndexOutOfBoundsException if permissive =>
             (index until schemaFields.length).foreach(ind => rowArray(ind) = null)
             Some(Row.fromSeq(rowArray))
-          case nfe: java.lang.NumberFormatException if dropMalformed =>
+          case _: java.lang.NumberFormatException |
+               _: IllegalArgumentException if dropMalformed =>
```
Comment (HyukjinKwon) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: Is this newline appropriate? I cannot find a proper document in [databricks/scala-style-guide](https://github.com/databricks/scala-style-guide).

**Comment by falaki** [comment]
Thanks @HyukjinKwon this looks good.

**Pull request closed without merging**